src/components/organisms/TableHeaderGroup/TableHeaderGroup.tsx

```diff
@@ -0,0 +1,128 @@
+import { useCallback, useState } from 'react'
+import { isEmpty, toString, size } from 'lodash'
+import { useTranslation } from 'react-i18next'
+import {
+  Table,
+  ColumnDef,
+  flexRender,
+  Header,
+  RowData,
+} from '@tanstack/react-table'
+import { DropDownOptions } from 'components/organisms/SelectionControlsInput/SelectionControlsInput'
+import Button, {
+  AppearanceTypes,
+  SizeTypes,
+  IconPositioningTypes,
+} from 'components/molecules/Button/Button'
+import { ReactComponent as SortingMore } from 'assets/icons/sorting_arrows.svg'
+import { ReactComponent as SortingLess } from 'assets/icons/sorting_less.svg'
+import TableColumnFilter from 'components/organisms/TableColumnFilter/TableColumnFilter'
+import classes from './styles.module.scss'
+interface HeaderGroupFunctions {
+  onSortingChange?: (value: string | boolean, columnId: string) => void
+  onColumnFiltersChange?: (filters: string[], columnId: string) => void
+}
+type HeaderGroupProps<TData> = {
+  table: Table<TData>
+} & HeaderGroupFunctions
+
+type ColumnMeta = {
+  meta?: {
+    size?: number | string
+    filterOption?: DropDownOptions[]
+    sortingOption?: (string | boolean)[]
+  }
+}
+type CustomColumnDef<TData> = ColumnDef<TData> & ColumnMeta
+
+type HeaderItemProps<TData> = {
+  hidden?: boolean
+  header: Header<TData, RowData>
+} & HeaderGroupFunctions
+
+const HeaderItem = <TData extends object>({
+  hidden,
+  header,
+  onSortingChange,
+  onColumnFiltersChange,
+}: HeaderItemProps<TData>) => {
+  const { t } = useTranslation()
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const { id, column } = header || {}
+  const { meta } = column.columnDef as CustomColumnDef<TData>
+  const filterOption = meta?.filterOption || []
+  const sortingOption = meta?.sortingOption || []
+  const value = sortingOption[currentIndex]
+  const newIndex = size(sortingOption) - 1 > currentIndex ? currentIndex + 1 : 0
+
+  const handleOnSorting = useCallback(() => {
+    if (onSortingChange) {
+      onSortingChange(value, id)
+      setCurrentIndex(newIndex)
+    }
+  }, [id, newIndex, onSortingChange, value])
+
+  if (hidden) return null
+
+  return (
+    <div className={classes.headingWrapper}>
+      {!isEmpty(sortingOption) && (
+        <Button
```

Comment:
**[blocker]**

While these smaller inline conditional rendering statements are not much of an issue, they should still not be used, if the underlying component has the "hidden" prop defined. Our Button has the "hidden" prop, so would be better to use that instead.